### ISSUE TITLE
Definition of background color to suppress gsplats of that color

### ISSUE THREAD

**Issue by simonbethke** [nerfstudio-project/nerfstudio#2842]
When capturing entire scenes outdoors, an overcast sky is preferable because one would not capture his own moving shadow. In such scene, looking up into the trees, those have a clear contrasat between branches and the sky which is entirely white (or at least light gray). When capturing trees like this, the splatfacto method will greate gsplats within the tree-branch-space that get the color of the sky. A leaf-less tree now would get filled with white splats.

**Proposed straight forward solution**
A simple solution would be to define a background color. When taking photos with an overcast sky, one would set this color to white and the algorithms should simply not create white splats (or cull them). For finetuning it would be helpful if the algorithm would know the up-direction to define in which directions sky is expected. For example if the horizon is visible and the capture takes place at a flat ground area, the sky can be max. 90° off the up-direction. Setting this would help, to still allow white reflections of the sky on ground-pieces.
To let the algorithm know what vector is up, I'd suggest to define that the first image for training should be leveled that the up-direction of the camera is up for the entire scene. 

**A better and more sophisticated solution**
A better (and more complicated) alternative would be, to use 2D-algorithms for masking the sky. Once the sky color around a certain area is known, that sky color (and also aliasing edges) should not create gaussians. Maybe even better, an optimization pass should subtract the sky color from those guassians making white gaussians transparent until culling them and gaussians which have a mixture of brown and white become entirely brown with 50% alpha.

**Additional context**
Here is a gsplat tree from the usual viewing angle:
![splat1](https://github.com/nerfstudio-project/nerfstudio/assets/11922599/6afb4a98-fd82-4359-b49e-7f680a6ef4ee)
And here it becomes clear that it has many white splats within the branches:
![splat2](https://github.com/nerfstudio-project/nerfstudio/assets/11922599/f85c1c2e-400c-4ad3-a755-6077dfdc93f6)
Not creating splats for skies will also help when trying to view a scene from above:
![splat3](https://github.com/nerfstudio-project/nerfstudio/assets/11922599/9e375262-5586-438c-a330-003a6368d64c)

**Comment by jaco001** [comment]
Only the way is mask utilities. There is a plenty cases that you need masks, not only white sky and tree. ;)

**Comment by simonbethke** [comment]
Right, masking would be a simple approach accomplishing this with available features. However, masks cannot help with half brown half white pixels. Edges of branches in 2D will still pick up white sky color.

**Comment by kerrj** [comment]
Sky behavior in splatting is definitely a shortcoming. We do have the option to specify a white or black background color in the method config, this might help your case some.

**Comment by jb-ye** [comment]
A general idea to address this issue is to use a pre-trained depth prediction model and incorporate its knowledge in a way in splatfacto training. How to do it exactly is not trivial though.

**Comment by simonbethke** [comment]
I think depth prediction algorithms might not be accurate enough. However, if the splatfacto method 'knows' that the skydome is white, it should be able to move white splats to infinity and keep the non-white splats.
